utils/s2n_blob.c

```diff
@@ -43,9 +44,10 @@ int s2n_blob_init(struct s2n_blob *b, uint8_t * data, uint32_t size)
 
 int s2n_blob_zero(struct s2n_blob *b)
 {
-    memset_check(b->data, 0, b->size);
-
-    return 0;
+    PRECONDITION_POSIX(s2n_blob_is_valid(b));
+    memset_check(b->data, 0, MAX(b->allocated, b->size));
```

Comment:
Does keeping this as `b->size` pass the tests?

Reply:
Nope, the HMAC proof still fails.

Reply:
I updated the spec for the failed proof and now the checks passed. Could you review it again?